libs/ml/include/ml/core/graph.hpp

```diff
@@ -18,20 +18,9 @@
 //------------------------------------------------------------------------------
 
 #include "ml/core/node.hpp"
-#include "ml/meta/ml_type_traits.hpp"
 #include "ml/ops/weights.hpp"
 #include "ml/state_dict.hpp"
```

Comment:
I think you can forward declare this and just include it in the source file?

Reply:
If I just forward declare StateDict, then I get this error:
error: implicit instantiation of undefined template 'fetch::ml::StateDict<fetch::math::Tensor<signed char, fetch::memory::SharedArray<signed char, 1> > >'
  virtual fetch::ml::StateDict<TensorType> StateDict();

Cannot figure out if it is possible to fix it without including "ml/state_dict.hpp"
                                           ^